mlflow/tracking/default_experiment/registry.py

```diff
@@ -62,14 +65,11 @@ def get_experiment_id():
     Returns:
         An experiment_id.
     """
-
-    experiment_id = DEFAULT_EXPERIMENT_ID
     for provider in _default_experiment_provider_registry:
         try:
             if provider.in_context():
-                experiment_id = provider.get_experiment_id()
-                break
+                return provider.get_experiment_id()
         except Exception as e:
             _logger.warning("Encountered unexpected error while getting experiment_id: %s", e)
 
-    return experiment_id
+    return DEFAULT_EXPERIMENT_ID if not is_databricks_uri(get_tracking_uri()) else None
```

Comment:
`DEFAULT_EXPERIMENT_ID` should not be used in Databricks. It can lead to a weird error message like `[View] on node with ID '0'`.